src/builtin_entity_parser.rs

```diff
@@ -267,6 +265,16 @@ mod test {
                 .collect_vec()
         );
 
+        assert_eq!(
+            vec![BuiltinEntityKind::Date],
+            parser
+                .extract_entities("Book me restaurant for tomorrow", Some(&[BuiltinEntityKind::Date]))
+                .unwrap()
+                .iter()
+                .map(|e| e.entity_kind)
+                .collect_vec()
+        );
```

Comment:
Could we also test that it works correctly with a filter containing `BuiltinEntityKind::Datetime` ? Something like:

```rust
assert_eq!(
    vec![BuiltinEntityKind::Datetime],
    parser
        .extract_entities("Book me a restaurant for tomorrow", Some(&[BuiltinEntityKind::Datetime]))
        .unwrap()
        .iter()
        .map(|e| e.entity_kind)
        .collect_vec()
);
```

Also, one of the use cases we want to address is to extract several dates or times in a sentence instead of a single date or time period:

```rust
assert_eq!(
    vec![BuiltinEntityKind::Time, BuiltinEntityKind::Time],
    parser
        .extract_entities("Book the meeting room from 10am to 11am", Some(&[BuiltinEntityKind::Time]))
        .unwrap()
        .iter()
        .map(|e| e.entity_kind)
        .collect_vec()
);
```

Reply:
✅